notes/gefs_forecasts.qmd

```diff
@@ -0,0 +1,183 @@
+              #dx = 0.5, dy = 0.5, # original resolution -- half-degree
+              dx = 0.1, 
+              dy = 0.1,
+              dt = "PT1H",
+              #dt = "PT3H",   # original resolution
+               aggregation = "mean", # will only aggregate if multiple ensemble members are viewed
+               resampling = "cubicspline"
+)
+```
+
+Now generate a dataframe of all the predictions.
+```{r}
+
+preds <- list()
+for(ens in names(gefs_cubes)){
+  gefs_cube <- gefs_cubes[[ens]]
+  x <- raster_cube(gefs_cube, v)  |> 
+         extract_geom(sf = azmet_pts)
+  p <- azmet_pts
+  p$FID <- rownames(azmet_pts)
+  y <- merge(p, x, by = "FID")
+  preds[[ens]] <- y
+}
+ens_forecast <- dplyr::bind_rows(preds, .id = 'ens')
+```
+
+
+Next step: long w/ summaries
+
+```{r}
+## create statistical summaries by station, variable, and time
+
+ens_stats <- ens_long %>% 
+  group_by(meta_station_name, var, time) %>% 
+  summarise(median = median(value),
+            min = min(value),
+            max = max(value),
+            ucl_95 = quantile(value, 0.975),
+            lcl_95 = quantile(value, 0.025),
+            ucl_99 = quantile(value, 0.995),
+            lcl_99 = quantile(value, 0.005)) %>% 
+  ungroup() %>% 
+  mutate(id = paste0(meta_station_name, var))
+# save above as csv
+readr::write_csv(ens_stats, 'ens_stats.csv')
+```
+
+```{r}
+## create a plot of ens_stats time vs. value, with color by station, and facet by variable
+library(ggplot2)
+
+a <- ggplot(data = ens_stats) +
+  geom_line(aes(x = time, y = median, color = meta_station_name)) +
+  geom_ribbon(aes(x = time, ymin = lcl_95, ymax = ucl_95, fill = meta_station_name), alpha = 0.25) +
+  geom_ribbon(aes(x = time, ymin = min, ymax = max, fill = meta_station_name), alpha = 0.05) +
+  facet_wrap(~var, ncol = 4, scales = 'free_y')
+# make plot above with wide aspect
+ggsave(a, width = 20, height = 10, filename = 'fig.pdf')
+
+# above plot but saved as one file per station, using a for loop
+plots <- list()
+for(stn in unique(ens_stats$meta_station_id)){
+  plots[[stn]] <- ggplot(data = ens_stats %>% filter(meta_station_name == stn)) +
+    geom_line(aes(x = time, y = median)) +
+    geom_ribbon(aes(x = time, ymin = lcl_95, ymax = ucl_95), alpha = 0.25) +
+    geom_ribbon(aes(x = time, ymin = min, ymax = max), alpha = 0.05) +
+    facet_wrap(~var, ncol = 4, scales = 'free_y')
+  ggsave(plots[[stn]], width = 20, height = 10, filename = paste0(stn, '.pdf'))
+}
+
+# with list above use cowplot to combine into one pdf
+library(cowplot)
+p <- plot_grid(plotlist = plots, ncol = 6, labels = names(plots))
+ggsave(p, width = 40, height = 20, filename = 'all.pdf')
+```
+
+```{r}
+```
+
+```{r}
+
+```
+
+
+Look at the data as a movie:
+- should add station points and maybe some county boundaries to this
+```{r}
+raster_cube(gefs_cube, v) |>
+  gdalcubes::select_bands("TMP") |>
+  animate( col = viridisLite::viridis, nbreaks=100,
+           fps=10, save_as = "temp.gif")
```

Comment:
Requires packge `gifski` which requires some system library that isn't in rocker/geospatial.  Cool idea, but probably won't implement in the qa/qc report.